### DISCUSSION TITLE
Wallbox lädt mit voller Leistung außerhalb der PV Erzeugung

### DISCUSSION THREAD

**Discussion by mh79hf** [camueller/SmartApplianceEnabler#622] [Allgemein]
Hallo Forum, Hallo Camueller,
ich habe immer mal wieder das Phänomen das die Wallbox Abends unbegreiflicher Weise mit maximaler Leistung
das Auto laden will. Im SAE habe ich "nur PV Überschuss" gewählt. Somit sollte das folgende Phänomen nicht auftreten. 

![image](https://github.com/user-attachments/assets/bbee905e-0904-4dd6-8fb1-e3e98b3598b3)

Gegen 17:30 Uhr hätte die Wallbox stoppen müssen. Stattdessen fängt sie an den Wagen mit maximaler Leistung zu versorgen. Die folge ist, das der Speicher leergezogen wird und der Netzbezug steigt. 

In den Logs finde ich die ensprechenden Startbefehle des Homemanagers :

![image](https://github.com/user-attachments/assets/cedb9a90-4f1a-48b0-b9e4-e485c6b6a7e0)

Dies wirft natürlich die Fragen auf warum meint der SHM das die BOX gestartet werden muss? Warum korrigiert er innerhalb von Sekunden die Leistung von 1380w auf 22KW?

Hat jemand eine Idee was dieses Verhalten erklären könnte bzw. wo nach ich noch schauen könnte?

Dies sind die Einstellungen im SHM für die Wallbox  : 
![image](https://github.com/user-attachments/assets/5b95b006-eaad-43e9-94bb-9f92be85eb76)

Vg Clyde

**Comment by camueller** [comment]
Das ist eine interessante Frage, warum der HM bei Dir die Leistung so massiv hochsetzt kurz nach dem Starten des Ladens. Das kann vermutlich nur SMA und die blocken meist, wenn sie höhren, dass externe Software im Spiel ist.

Bei mir ist das nicht der Fall, aber ich habe auch keinen Speicher im Haus, was es vermutlich für den HM einfach macht. Außerdem habe ich Netzbezug auf 0% stehe (wo Du 5% hast). Auch eine Wirkleistungseinspeisung-Begrenzung habe ich nicht mehr drin.

**Comment by mh79hf** [comment]
Hi, danke für die Antwort. Ich habe den SHM mal resetet und die 5% raus genommen. Ich hatte die drin, da SMA das empfiehlt.  Das hatte ich irgendwo in der Doku gelesen. Dachte es macht dem SHM es einfacher an die Grenzen zu gehen. 
Ich werde es erstmal versuchen über die Schaltzeiten des SAE einzugrenzen. So dass er erstmal nicht die ganze Nacht läd sollte es wieder auftreten.

Macht es Sinn die maximale Ladeleistung im SAE auf 7,2kw zu setzen? Die Wallbox kann 22KW 3PH. Das Auto kann eh nur 1 PH 7,2KW.
Andere Fahrzeuge werden hier zu 99% nicht geladen

Ich werde berichten  ... Danke

**Comment by mh79hf** [comment]
Hallo Forum, Hallo Camueller,
gestern hat mal wieder die Wallbox gelade, obwohl sie nicht sollte. Hierbei sind mir einige Dinge aufgefallen. Vielleicht hat jemand hierzu eine Erklärung. Vielleicht ist es auch ein Benutzerproblem :)

![image](https://github.com/user-attachments/assets/e1c7e7f2-2c2d-456a-87ba-c29a39ea1668)

Gegen 15Uhr habe ich im SAE den Ladevorgang mittels der Ampel händisch gestoppt. Gegen 16:00 Uhr startete der Ladevorgang wieder von selbst. 

1. Frage ) Kann ein Stop am SAE durch den SHM wieder überschrieben werden?

Der Ladevorgang endete gegen 22:00 Uhr.  Im SAE ist nur ein Zeitfenster bis 19:00 Uhr erlaubt. 

2. Frage ) Was kann bewirken das das Zeitfenster ingoriert wird? 

![image](https://github.com/user-attachments/assets/4048ea97-382c-4341-8660-f3258f2c6674)

 Viele Grüße,
 Cylde

**Comment by camueller** [comment]
zu 1) Wenn ein Ladevorgang durch Klick auf das rote Ampellicht gestoppt wird, wird dem SHM mitgeteilt, dass keine Schaltbefehle akzeptiert werden. Das gilt aber nur für die Dauer des aktuell aktiven Zeitfensters.

zu 2) Dafür müsstest Du die relevanten Teile des Log-Files vom SAE verfügbar machen. Darin ist enthalten, was der SAE dem SHM mitgeteilt hat und wie er darauf reagiert hat.

**Comment by mh79hf** [comment]
Hallo Forum, Hallo Camueller,
ich habe die Logs des 01-10-24 nach "F-08031979-000000000021" gefiltert und angehängt. Ich hoffe durch den Filter sind keine Informationen verloren gegangen.  Vielen Dank für's Prüfen.
[log.zip](https://github.com/user-attachments/files/17239775/log.zip)

 Vg Clyde

**Comment by camueller** [comment]
Also um 14:40 wurde das Fahrzeug von der Wallbox getrennt:
```
2024-10-01 14:40:52,433 DEBUG [Timer-0] d.a.s.c.e.ElectricVehicleCharger [ElectricVehicleCharger.java:331] F-08031979-000000000021-00: Vehicle state changed: previousState=CHARGING newState=VEHICLE_NOT_CONNECTED
```

Und um 15:15 wurde es erneut mit der Wallbox verbunden:
```
2024-10-01 15:16:53,692 DEBUG [Timer-0] d.a.s.c.e.ElectricVehicleCharger [ElectricVehicleCharger.java:331] F-08031979-000000000021-00: Vehicle state changed: previousState=VEHICLE_NOT_CONNECTED newState=CHARGING
```
Durch das erneute Verbinden des Fahrzeugs mit der Wallbox sind alle zuvor getroffenen Einschränkungen wieder zurückgesetzt.

**Comment by mh79hf** [comment]
Ok, Danke dir.  Aber das Zeifenster ( in dem Fall bis 19:00 Uhr ) hätte doch nicht ignoriert werden dürfen.

**Reply by camueller** [reply]
Wofür soll denn der Zeitplan überhaupt gut sein? Das sieht fast so aus, als wenn das zum Überschussladen sein soll. Dafür brauchst Du überhaupt keinen Zeitplan, das ist automatisch aktiv, wenn Du das Fahrzeug mit der Wallbox verbindest.

Du hast eine min. Energiemenge von 1380 Wh angegegeben, aber eigentlich wolltest Du vermutlich die min. Ladeleistung auf 1380W festlegen. Insofern ist das auch die falsche Stelle für diese Angabe.

**Comment by mh79hf** [comment]
Ich wollte damit verhindern, das der Wagen ungewollt die ganze Nacht durchläd. So wie oben in dem beschrieben Fall. 
So wie ich es verstanden hatte liegt es am SHM das eingeschaltet wurde. Da war mein Ziel zumindest manuell hier bis 19:00 Uhr zu begrenzen. 
Ich werde die Energiemengen da raus löschen. Du hattest recht. Ich hatte es mit der Ladeleistung verwechselt. Die aber ja schon in der Wallbox hinterlegt. 

Danke dir. Vg Clyde

**Comment by mh79hf** [comment]
Hallo Forum,
leider habe ich immer noch nicht meinen Go E Charger per SAE sauber am laufen. Der Charger fängt an zu laden wenn man Nachts nach Hause kommt und der SAE blinkt in grün als wenn PV Überschuss vorhanden wäre. Ich weiß erlich gesagt nicht mehr, wo ich noch sinnvoll nach dem Fehler suchen könnte. 

Hat jemand von euch diese Kombination sauber zum laufen bekommen?

Die Kommunikation per Shelly zu anderen Komponenten läuft deutlich besser.  Daher denke ich nicht das der SHM oder der SAE das Problem an sich sind. 

Hier ein Status von heute morgen 9:30 Uhr. 
![image](https://github.com/user-attachments/assets/fc9c095e-70e3-4fe1-8019-40ca5065294b)
PV Strom ist nicht genügend vorhanden.

![image](https://github.com/user-attachments/assets/c5e40e66-9aaa-4f0d-bfbb-08a97369fdc5)
Laut SHM sollte der Charger aus sein.

![image](https://github.com/user-attachments/assets/a019fe8e-9253-4d7e-bbe3-5d12e6df635e)
Die Wallbox blinkt grün und läd mit 6A

![image](https://github.com/user-attachments/assets/7dc773d3-8b3b-4cad-8342-16e9d6080af0)
Die Anzeige vom Charger zeigt, das er die ganze Nacht geladen hat.



![image](https://github.com/user-attachments/assets/081a5ab3-b70f-4e44-9078-891e6a62a88a)
![image](https://github.com/user-attachments/assets/2cff859e-93c3-45a3-a06e-e23424a99c8c)
![image](https://github.com/user-attachments/assets/402b39ae-7c6c-4b44-9d3c-ea580d0614d7)


Falls jemand eine Idee hat, würde ich mich freuen.

Vg Clyde

**Reply by camueller** [reply]
Poste doch nochmal ein Log, vielleicht findet sich ja diesmal mehr darin. Ggf. kann ich auch eine konkrete Frage an SMA weiterleiten.

**Comment by mh79hf** [comment]
Hallo camueller,
nett das du dir die Logs nochmal anschauen magst. Ich habe sie nach der Wallbox gefiltert. Also nur Zeilen, in der die ID vorkommt.
Ich hoffe das reicht. 

Viele Grüße,
Clyde

[rolling-2024-11-15_Filter.zip](https://github.com/user-attachments/files/17787199/rolling-2024-11-15_Filter.zip)
[rolling-2024-11-16Filter.zip](https://github.com/user-attachments/files/17787200/rolling-2024-11-16Filter.zip)

**Comment by mh79hf** [comment]
Im Ansteckzeitraum des Wagens ( so um 22:55 Uhr am 15.1.11)  sind negative Werte. Könnte das Probleme verursachen?
![image](https://github.com/user-attachments/assets/673aa1c2-2995-4d50-873b-f7755a9baae1)

**Reply by camueller** [reply]
Wie sagt man so schön: Garbage In - Garbage out. Bevor da Zeit für Fehlersuche investiert wird, würde ich zuerst dafür sorgen der Zähler sinnvolle Werte liefert.

**Reply by mh79hf** [reply]
Danke für den Hinweis. Ich habe exakt die Konfiguration eingehalten wie hier beschrieben  : https://github.com/camueller/SmartApplianceEnabler/blob/master/doc/GoeCharger_DE.md
Im GO e Charger kann man nichts einstellen. 

![image](https://github.com/user-attachments/assets/f043a853-dc43-457c-bed8-7f5f76549c70)

Wo könnte man noch nach dem Fehler suchen?

**Reply by camueller** [reply]
Vom go-eChanger existieren verschieden Hardware- und Firmware-Versionen. Ich verfolge das nicht genau, weil ich selbst eine andere Wallbox nutze. Wenn sich ein go-eCharger anders verhält als erwartet, muss man sich die API-Spezifikation des go-eCharger anschauen und vergleichen mit den Befehlen, die der SAE sendet auf Basis "seiner Konfiguration" für den go-eCharger. Letztlich ist das genau die Seite aus den SAE-Einstellungen für die Wallbox, die Du oben gepostet hast. Und die Konfiguration des Zähler im SAE. Beides musst Du (oder ein go-eCharger-User) mit der Spezifikation abgleichen, um festzustellen, ob sich etwas geändert hat.

**Comment by mh79hf** [comment]
ich stelle mal Testhalber auf "Leistung" um : 
![image](https://github.com/user-attachments/assets/6191ae63-c1ee-4c82-b7a2-05027b56131f)

**Reply by camueller** [reply]
"Testweise" ist in diesem Fall kein guter Ansatz zur Problemlösung: Du solltest schon wissen, ob Du in den Daten den Zählerstand oder die Leistung erwartest. Das kann man ja auch direkt sehen - ist ja HTTP. Das kannsu Du sogar im Web-Browser testen, indem Du die gleiche Seite des go-eCharger aufrufst, wie der SAE.

**Comment by mh79hf** [comment]
Natürlich würde ich schon dann auch schon den Extraktionspfad auf $.nrg11 mit umstellen. Sonst ist mir klar das da nichts bei raus kommen kann. 

GO E sagt zum "$.dws" folgendes : Geladene Energiemenge in Deka-Watt-Sekunden
                                                        Beispiel: 100'000 bedeutet, 1'000'000 Ws (= 277Wh = 0.277kWh)
                                                         wurden in **diesem** Ladevorgang geladen.

https://github.com/goecharger/go-eCharger-API-v1/blob/master/go-eCharger%20API%20v1%20DE.md

Wenn sich $.dws nur auf den aktuellen Ladevorgang beschränkt, wäre das doch nicht der richte Wert für den SAE, oder?

Vielleicht wäre "$.eto" besser? energy_total: Gesamt geladene Energiemenge in 0.1kWh
                                                                       Beispiel: 130 bedeutet 13kWh geladen

Vg Clyde

**Comment by camueller** [comment]
> Wenn sich $.dws nur auf den aktuellen Ladevorgang beschränkt, wäre das doch nicht der richte Wert für den SAE, oder?

Warum soll das nicht richtig sein? Das ist halt ein Zähler, der bei jedem Verbinden des Fahrzeugs mit der Wallbox wieder bei 0 startet. Der absolute Zählerwert ist doch dem SAE egal - er berechnet die Leistungs aus Zählerstandsdifferenzen und Zeitdifferenzen. Und nur die Leistung wird an den Sunny Home Manager gemeldet.

Das Berechnen der Leistung funktioniert nur sinnvoll (siehe SAE-Doku) , wenn der Zählerstand mit 5-Nachkomstellen geliefert wird. Das ist ja bei "$.eto" nicht der Fall. Alternativ kann man natürlich direkt die Leistung der Wallbox abfragen anstelle des Zählerstandes.

**Comment by mh79hf** [comment]
Ok ... danke für die Info. Ich teste mal mit $.nrg[11] und Leistung und prüfe ob ich dann immer noch negative Werte bekomme.
Eine andere Möglichkeit habe ich sonst nicht.
